Graph/test_main.py

```diff
@@ -0,0 +1,104 @@
+import pytest
+from main import Graph
+
+
+def test_add_edge():
+    graph = Graph()
+    graph.add_edge(1, 2)
+    graph.add_edge(2, 3)
+    assert graph.adjacency_list == {1: [2], 2: [1, 3], 3: [2]}
+
+
+def test_add_edge_string():
+    graph = Graph()
+    graph.add_edge(1, 2)
+    graph.add_edge(2, 3)
+    graph.add_edge("a", "b")
+    assert graph.adjacency_list == {
+        1: [2],
+        2: [1, 3],
+        3: [2],
+        "a": ["b"],
+        "b": ["a"],
+    }
+
+
+def test_add_edge_same_value():
+    graph = Graph()
+    graph.add_edge(1, 1)
+    assert graph.adjacency_list == {1: [1, 1]}
```

Comment:
Good test case.

But this exposes the problem that implementation is wrong here.

```suggestion
    assert graph.adjacency_list == {1: [1]}
```
We added only edge so there should only be one value in adjacency list of 1.